module/zfs/arc.c

```diff
@@ -3471,8 +3472,14 @@ arc_prune_async(int64_t adjust)
 			continue;
 
 		refcount_add(&ap->p_refcnt, ap->p_pfunc);
+		adjustbak = ap->p_adjust;
 		ap->p_adjust = adjust;
-		taskq_dispatch(arc_prune_taskq, arc_prune_task, ap, TQ_SLEEP);
+		if (taskq_dispatch(arc_prune_taskq, arc_prune_task,
+		    ap, TQ_SLEEP) == 0) {
+			ap->p_adjust = adjustbak;
```

Comment:
There's no need to preserve `adjust` in this function.  It's assigned to the ap->p_adjust and only valid while the callback is executing.  Multiple concurrent callback are prevented by the `refcount_count()` check above.  Go ahead and remove `adjustbak`.